tools/system_libs.py

```diff
@@ -700,6 +700,13 @@ def get_files(self):
     if shared.Settings.WASM_BACKEND:
       # See libc_extras below
       libc_files.append(shared.path_from_root('system', 'lib', 'libc', 'extras.c'))
+      # Include all the getenv stuff with the wasm backend. With fastcomp we
+      # still use JS because libc is a .bc file and we don't want to have a
+      # global constructor there for __environ, which would mean it is always
+      # included.
+      libc_files += files_in_path(
+          path_components=['system', 'lib', 'libc', 'musl', 'src', 'env'],
+          filenames=['__environ.c', 'getenv.c', 'putenv.c', 'setenv.c', 'unsetenv.c'])
```

Comment:
Can we do this instead by removing 'env' from blacklist above?  Then adding to blacklist_files accordingly?  Seems more in keeping with code here.

Reply:
There are some more files in that directory, though, like "init tls" and "init security" which we don't need. (what are those, I wonder..?)